src/lammpsio/data.py

```diff
@@ -121,14 +127,38 @@ def create(cls, filename, snapshot, atom_style=None):
             f.write(
                 f"LAMMPS {filename}\n\n"
                 f"{snapshot.N} atoms\n"
-                f"{num_types} atom types\n"
+                f"{snapshot.num_types} atom types\n"
                 f"{snapshot.box.low[0]} {snapshot.box.high[0]} xlo xhi\n"
                 f"{snapshot.box.low[1]} {snapshot.box.high[1]} ylo yhi\n"
                 f"{snapshot.box.low[2]} {snapshot.box.high[2]} zlo zhi\n"
             )
             if snapshot.box.tilt is not None:
                 f.write("{} {} {} xy xz yz\n".format(*snapshot.box.tilt))
 
+            if snapshot.has_bonds():
```

Comment:
Seeing below, I think these you just want to check like:
```suggestion
            if snapshot.bonds is not None:
```
because it is possible you may want to write a snapshot that has 0 bonds but does contain a bond type, for example. We should still use `has_bonds` when deciding whether to write the section though, as it should only be written if this number is nonzero.